pkg/cmd/pr/merge/merge.go

```diff
@@ -167,17 +168,45 @@ func mergeRun(opts *MergeOptions) error {
 			if err != nil {
 				return err
 			}
-			confirm, err := confirmSurvey()
+
+			allowEditMsg := (mergeMethod != api.PullRequestMergeMethodRebase) && !opts.BodyProvided
+
+			action, err := confirmSurvey(allowEditMsg)
 			if err != nil {
-				return err
+				return fmt.Errorf("unable to confirm: %w", err)
+			}
+
+			if action == shared.EditCommitMessageAction {
+				var editorCommand string
+				editorCommand, err = cmdutil.DetermineEditor(opts.Config)
+				if err != nil {
+					return err
+				}
+
+				msg, err := commitMsgSurvey(editorCommand)
+				if err != nil {
+					return err
+				}
+				opts.Body = msg
+				opts.BodyProvided = true
+
+				action, err = confirmSurvey(false)
+				if err != nil {
+					return fmt.Errorf("unable to confirm: %w", err)
+				}
 			}
-			if !confirm {
+			if action == shared.CancelAction {
 				fmt.Fprintln(opts.IO.ErrOut, "Cancelled.")
 				return cmdutil.SilentError
 			}
 		}
 
-		err = api.PullRequestMerge(apiClient, baseRepo, pr, mergeMethod, opts.Body)
+		var body *string
+		if opts.BodyProvided {
+			body = &opts.Body
```

Comment:
The reason this was necessary is because if we pass a blank string to `PullRequestMerge()` it will set the body to be a blank string leaving no way to use the default body message.